chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py

```diff
@@ -134,33 +141,142 @@ def fetch_device_list():
                 cmap(as_device), cfilter(filter_device), list)
 
 
-def add_to_ndp(xs, ys):
-    for x in xs:
-        for y in ys:
-            ndp.add_normalized_device(x, y)
+def parse_lvm_uuids(dm_uuid):
+    """
+    :param dm_uuid: device mapper uuid string (combined vg and lv with lvm prefix) from device_scanner
+    :return: tuple of vg and lv UUIDs
+    """
+    lvm_pfix = 'LVM-'
+    uuid_len = 32
+
+    if (not dm_uuid.startswith(lvm_pfix)) or (len(dm_uuid) != (len(lvm_pfix) + (uuid_len * 2))):
```

Comment:
What case is there a string that starts with `LVM-` but doens't have the correct length?

Reply:
should never happen, will amend